Fix Makefile issues found by `make --shuffle` (GNU Make 4.4+)

**Newsboat version (copy the output of `newsboat -v` or the first line of `git show`)**: Debian package 2.36-1

(Assumed) steps to reproduce the issue (I didn't try them because I don't have GNU Make 4.4 yet):

1. run `make --shuffle=3554967441 -jN`, presumably with N>1

GNU Make [recently gained ability to randomly re-order goals](https://lists.gnu.org/archive/html/bug-make/2022-02/msg00042.html), which introduces even more non-determinism than a mere `-jN`. That exposes missing dependencies between targets.

This functionality [got released with GNU Make 4.4](https://lists.gnu.org/archive/html/info-gnu/2022-10/msg00008.html), and a Debian Developer Santiago Vila [rebuilt the Debian archive with that option enabled](https://people.debian.org/~sanvila/make-shuffle/).

Newsboat 2.36 [failed to build](https://people.debian.org/~sanvila/make-shuffle/build-logs/newsboat_2.36-1_amd64-20240909T111415.334Z) with the following message:

```
...

g++ -std=c++14 -O2 -ggdb -Iinclude -Istfl -Ifilter -I. -Irss -I/<<PKGBUILDDIR>>/target/cxxbridge/ -Werror -Wall -Wextra -Wunreachable-code -DLOCALEDIR='"/usr/share/locale"' -DCATCH_AMALGAMATED_CUSTOM_MAIN -I/usr/include/x86_64-linux-gnu -I/usr/include/p11-kit-1  -I/usr/include/libxml2  -I/usr/include/json-c  -D_DEFAULT_SOURCE -D_XOPEN_SOURCE=600  -g -O2 -ffile-prefix-map=/<<PKGBUILDDIR>>=. -fstack-protector-strong -fstack-clash-protection -Wformat -Werror=format-security -fcf-protection -Wdate-time -D_FORTIFY_SOURCE=2 -MD -MP -MF .deps/newsboat.cpp -o newsboat.o -c newsboat.cpp
newsboat.cpp:23:10: fatal error: xlicense.h: No such file or directory
   23 | #include "xlicense.h"
      |          ^~~~~~~~~~~~
compilation terminated.
make[2]: *** [Makefile:178: newsboat.o] Error 1 shuffle=3554967441
make[2]: Leaving directory '/<<PKGBUILDDIR>>'
dh_auto_build: error: make -j1 "INSTALL=install --strip-program=true" CARGO_HOME=debian/cargo_home prefix=/usr all returned exit code 2
make[1]: *** [debian/rules:20: override_dh_auto_build] Error 25 shuffle=3554967441
make[1]: Leaving directory '/<<PKGBUILDDIR>>'
make: *** [debian/rules:12: binary] Error 2 shuffle=3554967441
```

([gzipped copy of the full log for posterity](https://github.com/user-attachments/files/17906746/newsboat_2.36-1_amd64-20240909T111415.334Z.gz))

Our Makefile clearly states that newsboat (the binary) depends on xlicense.h, so I don't understand what's going on here.

--- a/Makefile
+++ b/Makefile
@@ -185,6 +185,9 @@ target/cxxbridge/libnewsboat-ffi/src/%.rs.h: $(NEWSBOATLIB_OUTPUT)
 $(CPP_DEPS_SUBDIRS):
 	$(MKDIR) $@
 
+# Make sure xlicense.h is generated before building newsboat.cpp.
+newsboat.o: xlicense.h
+
 %.o: %.cpp # Cancel default rule for C++ code
 %.o: %.cpp | $(NEWSBOATLIB_OUTPUT) $(STFL_HDRS) $(CPP_DEPS_SUBDIRS)
 	$(CXX) $(CXXFLAGS) -MD -MP -MF $(addprefix .deps/,$<) -o $@ -c $<

